src/ulid.cr

```diff
@@ -19,6 +19,61 @@ module ULID
     encode_time(seed_time, TIME_LEN) + encode_random(RANDOM_LEN)
   end
 
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => true
+  # ```
+  def valid?(ulid : String) : Bool
+    # Incorrect length && Invalid chars
+    ulid.size == TIME_LEN + RANDOM_LEN && ulid.upcase.chars.all? &.in?(ENCODING)
+  end
+
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid!("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => nil (or exception)
+  # ```
+  def validate!(ulid : String) : Nil
+    # Incorrect length
+    raise IncorrectLength.new() unless ulid.size == TIME_LEN + RANDOM_LEN
+    # Invalid chars
+    raise InvalidChars.new() unless ulid.upcase.chars.all? &.in?(ENCODING)
+
+    nil
+  end
```

Comment:
can just return void here

Reply:
@dukeraphaelng [`Nil` is more idiomatic for this case](https://crystal-lang.org/reference/syntax_and_semantics/return_types.html#nil-return-type).

@GabFitzgerald you can drop the `nil` on L45 though. As you've specified the method type as returning `Nil` it always provide that, regardless of what's in the method body.